astropy/cosmology/flrw/base.py

```diff
@@ -1658,8 +1658,8 @@ def nonflat(self: _FlatFLRWMixinT) -> _FLRWT:
         # Make new instance, respecting args vs kwargs
         inst = self.__nonflatclass__(*ba.args, **ba.kwargs)
         # Because of machine precision, make sure parameters exactly match
-        for n in inst._parameters_all:
-            object.__setattr__(inst, "_" + n, getattr(self, n))
+        inst.__dict__.update(inst.parameters)
```

Comment:
I'm not quite sure I follow - I thought the problem was that you were still setting the private attribute here.

Anyway, as said, this was clearly a bit missing in the previous PR, so am fine with just getting it in!

Reply:
The private attributes "_Tcmb0" etc no longer exist, so `__setattr__` has to be updated to `__dict__[k] = v`.
The `.parameters` and `._derived_parameters` attributes are dicts of (k, v) for all k in `_parameters_all`.

Reply:
Yes, that makes sense - but why is this related to integer/floating point? Wouldn't a test here just check that the attributes are properly updated when making a `nonflat` instance?

Reply:
They should have already transferred correctly from the non-flat to flat class in the previous line:
`inst = self.__nonflatclass__(*ba.args, **ba.kwargs)`
However in practice, e.g. if there's any math like `Ode0 = 1 - x` done in the flat class' initialization, there are sometimes small mismatches in the values. This forces the parameters to be the same.

Reply:
Ah, good, so that means the tests would normally pass indeed, and the "bug" here was that private attributes were set that are obviously no longer needed.

With that, I think we should just merge this; I don't see how you are going to write a good test for it...